pyiron_base/jobs/job/extension/server/generic.py

```diff
@@ -552,58 +562,28 @@ def view_queues():
 
     def to_dict(self):
         server_dict = self._type_to_dict()
-        server_dict.update(
-            {
-                "user": self._user,
-                "host": self._host,
-                "run_mode": self.run_mode.mode,
-                "queue": self.queue,
-                "qid": self._queue_id,
-                "cores": self.cores,
-                "threads": self.threads,
-                "new_h5": self.new_hdf,
-                "run_time": self.run_time,
-                "memory_limit": self.memory_limit,
-                "accept_crash": self.accept_crash,
-            }
-        )
-        if len(self.additional_arguments) > 0:
-            server_dict["additional_arguments"] = self.additional_arguments
-        if self._gpus is not None:
-            server_dict["gpus"] = self._gpus
-        if self._environment_name is not None:
-            server_dict["conda_environment_name"] = self._environment_name
-        if self._environment_path is not None:
-            server_dict["conda_environment_path"] = self._environment_path
+        self._data.run_mode = self._run_mode.mode
+        server_dict.update(self._data.__dict__)
```

Comment:
```suggestion
        server_dict.update(asdict(self._data))
```
This requires `from dataclasses import asdict` in the beginning of the file.